test/jdk/java/net/URLConnection/RequestProperties.java

```diff
@@ -79,4 +82,29 @@ private static boolean hasFtp() {
             return false;
         }
     }
+
+    /**
+     * Test that {@link URLConnection#setRequestProperty(String, String)} throws
+     * an {@link IllegalStateException} when already connected
+     *
+     * @throws Exception
+     */
+    private static void testSetRequestPropIllegalStateException() throws Exception {
+        final URL url = Path.of(System.getProperty("java.io.tmpdir")).toUri().toURL();
+        final URLConnection conn = url.openConnection();
+        conn.connect();
+        try {
+            conn.setRequestProperty("foo", "bar");
```

Comment:
While here, and not directly related to this change, it would be good to add coverage for the other xxxRequestPropertXXX methods - to ensure that they also throw the correction exception when connected.

Reply:
Thank you for the review, Chris.

I've updated this PR to include these inputs into the test case.